src/Apps/Settings/Routes/EditProfile/Components/SettingsEditProfileFields.tsx

```diff
@@ -257,7 +230,33 @@ const SettingsEditProfileFields: React.FC<SettingsEditProfileFieldsProps> = ({
                 <Flex alignItems="center">
                   <CheckCircleIcon fill="black60" mr={0.5} />
                   <Clickable
-                    onClick={handleIDVerification}
+                    onClick={async () => {
+                      try {
+                        await submitVerifyIDMutation({
+                          variables: { input: {} },
+                          rejectIf: res => {
+                            return res.sendIdentityVerificationEmail
+                              ?.confirmationOrError?.mutationError
+                          },
+                        })
+
+                        sendToast({
+                          variant: "success",
+                          message: `ID verification link sent to ${
+                            me?.email ?? ""
```

Comment:
I think this is a good check to catch any unknown edge cases, but is there ever a scenario where we would expect to not have an email here?

Reply:
If this is due to type errors (don't think it is, but heads up), you can always use 
```
me?.email as string
```
to avoid using `any`

Reply:
@mc-jones good point. I double-checked and there are no such scenarios on `User`. Removed chaining and coalescing in https://github.com/artsy/force/pull/12153/commits/db39b0601aaeec4a9c1e6e468c96eb01c08feaa3

@damassi no type errors. thanks for showing an alternative tho!